src/components/generate/hooks.tsx

```diff
@@ -0,0 +1,106 @@
+import { useState, useMemo, useEffect } from 'react';
+
+export function useHandleSelection() {
```

Comment:
Seeing how you manage all the custom hooks was really helpful! Got to know more about useEncapsulation. (https://kyleshevlin.com/use-encapsulation)